controller/controller.go

```diff
@@ -1020,48 +1090,48 @@ func (ctrl *ProvisionController) provisionClaimOperation(claim *v1.PersistentVol
 	if err == nil && volume != nil {
 		// Volume has been already provisioned, nothing to do.
 		glog.Info(logOperation(operation, "persistentvolume %q already exists, skipping", pvName))
-		return nil
+		return ProvisioningFinished, nil
 	}
 
 	// Prepare a claimRef to the claim early (to fail before a volume is
 	// provisioned)
 	claimRef, err := ref.GetReference(scheme.Scheme, claim)
 	if err != nil {
 		glog.Error(logOperation(operation, "unexpected error getting claim reference: %v", err))
-		return nil
+		return ProvisioningNoChange, nil
```

Comment:
y NoChange here?

Reply:
Because I don't know what previous `ProvisionExt()` returned. I know it's very unlikely that `GetReference` succeeded once, `ProvisionExt()` returned `Background` and second `GetReference` fails, but I prefer to be on the safer side.